tests/components/multi_select_state.spec.js

```diff
@@ -205,9 +215,51 @@ describe("withMultiSelectState", () => {
   test("componentWillReceiveProps will not call setState when selectedItems is the same", () => {
     const state1 = [ITEM_3];
     const ConditionalComponent = withMultiSelectState(CustomComponent);
-    const wrapper = shallow(<ConditionalComponent selectedItems={state1} />);
+    const wrapper = shallow(<ConditionalComponent selectedItems={state1} />, {disableLifecycleMethods: true});
     wrapper.setProps({ selectedItems: state1 });
     wrapper.update();
     expect(wrapper.state().selectedItems).toBe(state1);
   });
+
+    test("can select with shift next items", () => {
+        const ConditionalComponent = withMultiSelectState(CustomComponent);
+        items.push(ITEM_4);
+        const wrapper = shallow(<ConditionalComponent items={items} />, {disableLifecycleMethods: true});
+        EVENT.shiftKey = true;
+        wrapper.props().selectItem(EVENT, ITEM_1.id);
+        wrapper.update();
+        expect(wrapper.prop("selectedItems")).toEqual([ITEM_1]);
+        expect(wrapper.state("firstItemShiftSelected")).toEqual(ITEM_1.id);
+        wrapper.props().selectItem(EVENT, ITEM_4.id);
+        wrapper.update();
+        expect(wrapper.prop("selectedItems")).toEqual([ITEM_1, ITEM_2, ITEM_3, ITEM_4]);
+    });
+
+    test("can select with shift previous items", () => {
+        const ConditionalComponent = withMultiSelectState(CustomComponent);
+        items.push(ITEM_4);
+        const wrapper = shallow(<ConditionalComponent items={items} />, {disableLifecycleMethods: true});
+        EVENT.shiftKey = true;
```

Comment:
lets avoid these mutations and create an event with shiftKey: true.
You're creating a side effect with doing this in tests which may break tests in the future.